source/operators/captions/webcaptions.py

```diff
@@ -0,0 +1,1016 @@
+headers = {"Content-Type": "application/json"}
+dataplane = DataPlane()
+
+mie_config = json.loads(os.environ['botoConfig'])
+config = config.Config(**mie_config)
+translate_client = boto3.client('translate', config=config)
+polly = boto3.client('polly', config=config)
+
+class WebCaptions:
+    def __init__(self, operator_object):
+        """
+        :param event: The event passed in to the operator
+
+        """
+        print("WebCaptions operator_object = {}".format(operator_object))
+        self.operator_object = operator_object
+
+        try:
+            self.transcribe_operator_name = "TranscribeVideo"
+            self.workflow_id = operator_object.workflow_execution_id
+            self.asset_id = operator_object.asset_id
+            self.marker = "<span>"
+            self.contentType = "text/html"
+            self.existing_subtitles = False
+            if "SourceLanguageCode" in self.operator_object.configuration:
+                self.source_language_code = self.operator_object.configuration["SourceLanguageCode"]
+                self.operator_name_with_lang = self.operator_object.name+"_"+self.source_language_code
+            if "TargetLanguageCode" in self.operator_object.configuration:
+                self.target_language_code = self.operator_object.configuration["TargetLanguageCode"]
+            if "ExistingSubtitlesObject" in self.operator_object.configuration:
+                self.existing_subtitles_object = self.operator_object.configuration["ExistingSubtitlesObject"]
+                self.existing_subtitles = True
+        except KeyError as e:
+            self.operator_object.update_workflow_status("Error")
+            self.operator_object.add_workflow_metadata(WebCaptionsError="No valid inputs {e}".format(e=e))
+            raise MasExecutionError(operator_object.return_output_object())
+
+    def WebCaptionsOperatorName(self, language_code=None, source=""):
+
+        # Shouldn't assume WebCaptions operator is WebCaptions, maybe pass it in the configuration?
+        print("WebCaptionsOperatorName {}, {}".format(language_code, source))
+
+        operator_name = "WebCaptions"+source
+
+        if language_code != None:
+            return operator_name+"_"+language_code
+
+        try:
+            name = operator_name+"_"+self.source_language_code
+        except KeyError:
+            self.operator_object.update_workflow_status("Error")
+            self.operator_object.add_workflow_metadata(WebCaptionsError="Missing language code for WebCaptions {e}".format(e=e))
+            raise MasExecutionError(self.operator_object.return_output_object())
+
+        print("WebCaptionsOperatorName() Name {}".format(name))
+        return name
+
+    def CaptionsOperatorName(self, language_code=None):
+
+        # Shouldn't assume WebCaptions operator is WebCaptions, maybe pass it in the configuration?
+        operator_name = "Captions"
+
+        if language_code != None:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `==` and `!=` operators use the compared objects&#39; ` __eq__` method to test if they are equal. To check if an object is a singleton,such as `None`, we recommend that you use the `is` identity comparison operator.

[Learn more](https://docs.python.org/3/reference/expressions.html#is)